app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt

```diff
@@ -460,15 +478,30 @@ fun ParkingDetailsScreen(
                                   text = stringResource(R.string.card_screen_user),
                                   style = MaterialTheme.typography.bodyMedium,
                                   color = MaterialTheme.colorScheme.onBackground)
-                              Text(
-                                  text =
-                                      if (userViewModel.selectedParkingOwner.value != null)
-                                          userViewModel.selectedParkingOwner.value
-                                              ?.public
-                                              ?.username!!
-                                      else stringResource(R.string.undefined_username),
-                                  style = MaterialTheme.typography.bodyMedium,
-                                  color = MaterialTheme.colorScheme.onSurface)
+                              if (range.color == stringResource(R.string.rainbow_text_color)) {
+                                Text(
+                                    text =
+                                        stringResource(
+                                            R.string.display_user_tag_format,
+                                            range.symbol,
+                                            level,
+                                            ownerUsername),
+                                    style = MaterialTheme.typography.bodyMedium,
+                                    color =
+                                        MaterialTheme.colorScheme
+                                            .onSurface // TODO rainbow text color
```

Comment:
I see that there are way too many places in the code with an if else on the color argument. Here is my suggestion:
You create a composable in atoms called DisplayText that takes as argument a composable Text and a string representing the color tag. DisplayText will be the one in charge of processing the string input and discriminate betweeen "rainbow" or a hex color tag and this will avoid having and If ("rainbow") ... else ... in all three files that display the username.. I don't know if that was clear. If it wasn't, please let me know. It's ok if for now the color isn't displayed as rainbow. The real problem is that this if else makes the code hard to read and maintain

Reply:
Yeah, I was also thinking something along those lines but you beat me to it. It's a Composable that's often repeated in code so it makes sense to use an Atom for this, w/ the only "varying" factor being the color

Reply:
Ok it's a bit too complicated for no reason to do that. Because the texts are from different instances (whether material3 or atoms or idk what else  even...), so their arguments differ. The parameters are different in each file.
And instead I ask you to accept my counter-offer, which is that i place the if else condition inside the color argument itself, reducing code size by 2 everywhere. That's already good enough.

Reply:
Unfortunate that you couldn't get our suggestion to work, but at the same time I understand that it's difficult and it's not really this PR's objective anyway. I'm okay w/ your suggestion, how about you @Armetiis?

Reply:
I think we can keep this for now. I suggested that because I was unsure of how complicated it would actually be to make the text rainbow colored. I doubt it's just a "Color" argument. But this is not a concern for now. This can be addressed when actually implementing the rainbow color, as that's when it will be possible to assess if it's necessary or not.

For the moment this is perfect, as the main concern to me was the heavy code duplication.